Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/AadhaarVerification.hs

```diff
@@ -339,3 +365,126 @@ checkForDuplicacy :: DbHash -> Flow ()
+  (blocked, driverDocument) <- DVRC.getDriverDocumentInfo person
+  when blocked $ throwError AccountBlocked
+  transporterConfig <- SCTC.findByMerchantOpCityId person.merchantOperatingCityId (Just (DriverId (cast person.id))) >>= fromMaybeM (TransporterConfigNotFound person.merchantOperatingCityId.getId)
+  case transporterConfig.allowDuplicateAadhaar of
+    Just False -> do
+      aadhaarHash <- getDbHash req.aadhaarNumber
+      aadhaarInfoList <- QAadhaarCard.findAllByEncryptedAadhaarNumber (Just aadhaarHash)
+      when (length aadhaarInfoList > 1) $ throwError AadhaarAlreadyLinked
+      aadhaarPersonDetails <- Person.getDriversByIdIn (map (.driverId) aadhaarInfoList)
+      let getRoles = map (.role) aadhaarPersonDetails
+      when (person.role `elem` getRoles) $ throwError AadhaarAlreadyLinked
+    _ -> pure ()
+  whenJust mbMerchant $ \merchant -> do
+    unless (merchant.id == person.merchantId) $ throwError (PersonNotFound personId.getId)
+  aadhaarInfo <- QAadhaarCard.findByPrimaryKey person.id
+  whenJust aadhaarInfo $ \aadhaarInfoData -> do
+    when (aadhaarInfoData.verificationStatus == Documents.VALID) $ throwError AadhaarAlreadyLinked
+  image1 <- getImage req.aadhaarFrontImageId
+  image2 <- case req.aadhaarBackImageId of
+    Just backImageId -> do
+      image <- getImage backImageId
+      return $ Just image
+    Nothing -> return Nothing
+  let extractReq =
+        Verification.ExtractAadhaarImageReq
+          { image1 = image1,
+            image2 = image2,
+            driverId = person.id.getId,
+            consent = if req.consent then "yes" else "no"
+          }
+  resp <- Verification.extractAadhaarImage person.merchantId merchantOpCityId extractReq
+  _ <- case resp.extractedAadhaar of
+    Just extractedAadhaarData -> do
+      let extractedAadhaarOutputData = extractedAadhaarData.extraction_output
+      let extractedAadhaarNumber = removeSpaceAndDash <$> extractedAadhaarOutputData.id_number
+      unless (extractedAadhaarNumber == Just req.aadhaarNumber) $
+        throwImageError (Id req.aadhaarFrontImageId) $
+          ImageDocumentNumberMismatch
+            (maybe "null" maskText extractedAadhaarNumber)
+            (maskText req.aadhaarNumber)
+      when (isJust transporterConfig.validNameComparePercentage) $
+        DVRC.validateDocument person.merchantId merchantOpCityId person.id extractedAadhaarOutputData.name_on_card extractedAadhaarOutputData.date_of_birth Nothing ODC.AadhaarCard driverDocument
+      aadhaarCard <- makeAadhaarCardEntity person.id extractedAadhaarOutputData req
+      QAadhaarCard.upsertAadhaarRecord aadhaarCard
+      pure extractedAadhaarData
+    Nothing -> throwImageError (Id req.aadhaarFrontImageId) ImageExtractionFailed
+  case person.role of
+    Person.FLEET_OWNER -> do
+      encryptedAadhaarNumber <- encrypt req.aadhaarNumber
+      QFOI.updateAadhaarImage (Just encryptedAadhaarNumber) (Just req.aadhaarFrontImageId) req.aadhaarBackImageId person.id
```

Comment:
here, don't you need to handle also `FLEET_BUSINESS` role?
Because late you enable both `FLEET_OWNER` and `FLEET_BUSINESS`
```
    role | isFleetOwnerRole role -> do
      enableFleet <- DFR.enableFleetIfPossible person.id adminApprovalRequired (DFR.castRoleToFleetType person.role)
```